extensions/amp-ad/0.1/amp-ad-3p-impl.js

```diff
@@ -240,6 +243,7 @@ export class AmpAd3PImpl extends AMP.BaseElement {
 
   /** @override  */
   unlayoutCallback() {
+    this.layoutPromise_ = null;
     if (!this.iframe_) {
```

Comment:
I only add one line to reset `layoutPromise_` for a new layout. otherwise I don't see why `unlayoutCallback()` logic should change.